File: test_javaPrj_01/src/programmers/lv0/다항식더하기.java
package src.programmers.lv0;

public class 다항식더하기 {
    private 다항식더하기(){
        try{
            Solution();
        }catch(Exception e){
            e.printStackTrace();
        }
    }

    @SuppressWarnings("unused")
    public void Solution() throws Exception{

        String polynomial = "3x + 7 + x + 2";

        System.out.println(polynomial.replaceAll(" ", ""));
        String[] array = polynomial.replaceAll(" ", "").split("[+]");

        
        int monomial = 0;
        int constant = 0;
        for(String element : array){
            if(element.indexOf("x") > -1){
                String str = element.replaceAll("x", "");
                monomial += str.length() > 0 ? Integer.valueOf(str) : 1;
            }else{
                constant += Integer.valueOf(element);
            }
        }
        
        String answer = "";
        if(monomial > 0){
            answer += monomial > 1 ? monomial + "x" : "x";
        }
        if(constant > 0){
            if(answer.length() > 0){
                answer += " + ";
            }
            answer += constant;
        }
        System.out.println(answer);
        // return monomial+"x + "+constant;

    }

    public static void main(String[] args) {
        new 다항식더하기();
    }
}

File: test_javaPrj_01/src/programmers/lv0/다항식더하기.py


import re
class 다항식더하기:

    def __init__(self, polynomial:str) -> None:

        print(  [str.strip() for str in polynomial.split('+')]  )

        monominal = 0
        constant = 0

        for element in [e.strip() for e in polynomial.split('+')]:
            
            if element.find('x') > -1:
                
                data = re.sub('x', '', element)
                monominal += int(data) if len(data) > 0 else 1

                
            else:
                constant += int(element)
        

        answer = ''
        if monominal > 0:
            answer += str(monominal) + 'x' if monominal > 1 else 'x'
        
        if constant > 0:
            if len(answer) > 0:
                answer += ' + '
            answer += str(constant)
        
        print(answer)

        

다항식더하기('3x + 7 + x')
